src/main/java/org/takes/facets/auth/codecs/CcAES.java

```diff
@@ -0,0 +1,167 @@
+ * SOFTWARE.
+ */
+package org.takes.facets.auth.codecs;
+
+import java.io.IOException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import lombok.EqualsAndHashCode;
+import org.takes.facets.auth.Identity;
+
+/**
+ * AES codec which support 128 bits key.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Jason Wong (super132j@yahoo.com)
+ * @version $Id$
+ * @since 0.13.8
+ */
+@EqualsAndHashCode(of = { "origin", "secret", "ivbytes" })
+public final class CcAES implements Codec {
+
+    /**
+     * The algorithm used.
+     */
+    private static final transient String ALGORITHM = "AES/CBC/PKCS5PADDING";
+
+    /**
+     * Default block size (128 bits) in bytes.
+     */
+    private static final transient int BLOCK = 16;
+
+    /**
+     * Original codec.
+     */
+    private final transient Codec origin;
+
+    /**
+     * Initialization Vector.
+     */
+    private final transient byte[] ivbytes;
+
+    /**
+     * The AES secret key object.
+     */
+    private final transient SecretKey secret;
+
+    /**
+     * Ctor.
+     * @param codec Original codec
+     * @param key The encryption key
+     */
+    public CcAES(final Codec codec, final byte[] key) {
+        if (key.length != BLOCK) {
+            throw new IllegalArgumentException(
+                    "Key length must be in 128-bit."
+            );
+        }
+        this.origin = codec;
+        final byte[] passcode = new byte[key.length];
+        System.arraycopy(key, 0, passcode, 0, key.length);
+        final SecureRandom random = new SecureRandom();
+        this.ivbytes = new byte[BLOCK];
+        random.nextBytes(this.ivbytes);
+        this.secret = new SecretKeySpec(passcode, "AES");
+    }
+
+    @Override
+    public byte[] encode(final Identity identity) throws IOException {
+        return this.encrypt(this.origin.encode(identity));
+    }
+
+    @Override
+    public Identity decode(final byte[] bytes) throws IOException {
+        return this.origin.decode(this.decrypt(bytes));
+    }
+
+    /**
+     * Encrypt the given bytes using AES.
+     *
+     * @param bytes Bytes to encrypt
+     * @return Encrypted byte using AES algorithm
+     * @throws IOException for all unexpected exceptions
+     */
+    private byte[] encrypt(final byte[] bytes) throws IOException {
+        try {
+            final Cipher cipher = Cipher.getInstance(ALGORITHM);
+            final AlgorithmParameterSpec spec =
+                    new IvParameterSpec(this.ivbytes);
```

Comment:
@super132 we can init decrypt/encrypt ciphers in Ctor, no need to create this each time.